script/integration/containerd/entrypoint.sh

```diff
@@ -101,36 +122,81 @@ if [ "${OK}" != "ok" ] ; then
 fi
 
 ############
-# Tests for mirror config
+# Tests for refreshing and mirror
+echo "Testing refreshing and mirror..."
+
+reboot_containerd
+echo "Getting image with normal snapshotter..."
+ctr-remote images pull --user "${DUMMYUSER}:${DUMMYPASS}" "${REGISTRY_HOST}:5000/alpine:stargz"
+ctr-remote run --rm "${REGISTRY_HOST}:5000/alpine:stargz" test tar -c /usr > /usr_dummy_org.tar
 
-reboot_containerd --log-level debug --config=/etc/containerd/config.toml
+echo "Getting image with stargz snapshotter..."
 PULL_LOG=$(mktemp)
-echo "Checking if mirroring is in effect"
-ctr-remote images rpull --plain-http "${REGISTRY_ALT_HOST}:5000/alpine:stargz" | tee "${PULL_LOG}"
+ctr-remote images rpull --user "${DUMMYUSER}:${DUMMYPASS}" "${REGISTRY_HOST}:5000/alpine:stargz" | tee "${PULL_LOG}"
 if ! isServedAsRemoteSnapshot "${PULL_LOG}" ; then
     echo "Failed to serve all layers as remote snapshots"
     exit 1
 fi
 
+REGISTRY_HOST_IP=$(getent hosts "${REGISTRY_HOST}" | awk '{ print $1 }')
+REGISTRY_ALT_HOST_IP=$(getent hosts "${REGISTRY_ALT_HOST}" | awk '{ print $1 }')
+
+echo "Disabling source registry and check if mirroring is working for stargz snapshotter..."
+iptables -A OUTPUT -d "${REGISTRY_HOST_IP}" -j DROP
+iptables -L
+ctr-remote run --rm --snapshotter=stargz "${REGISTRY_HOST}:5000/alpine:stargz" test tar -c /usr > /usr_dummy_mirror.tar
+iptables -D OUTPUT -d "${REGISTRY_HOST_IP}" -j DROP
+
+echo "Disabling mirror registry and check if refreshing works for stargz snapshotter..."
+iptables -A OUTPUT -d "${REGISTRY_ALT_HOST_IP}" -j DROP
+iptables -L
+ctr-remote run --rm --snapshotter=stargz "${REGISTRY_HOST}:5000/alpine:stargz" test tar -c /usr > /usr_dummy_refresh.tar
+iptables -D OUTPUT -d "${REGISTRY_ALT_HOST_IP}" -j DROP
+
+echo "Disabling all registries and running container should fail"
+iptables -A OUTPUT -d "${REGISTRY_HOST_IP}","${REGISTRY_ALT_HOST_IP}" -j DROP
+iptables -L
+if ctr-remote run --rm --snapshotter=stargz "${REGISTRY_HOST}:5000/alpine:stargz" test tar -c /usr > /usr_dummy_fail.tar ; then
+    echo "All registries are disabled so this must be failed"
+    exit 1
+else
+    echo "Successfully failed running container"
```

Comment:
wut

Reply:
"Successfully failed" -> "Failed as expected"?

Reply:
Sorry for the bad english:pray: Yes I meant "Failed as expected".